app/controllers/stripe-setup/government-entity-document/post.controller.js

```diff
@@ -0,0 +1,95 @@
+const paths = require('../../../paths')
+const formatAccountPathsFor = require('../../../utils/format-account-paths-for')
+
+const GOVERNMENT_ENTITY_DOCUMENT_FIELD = 'government-entity-document'
+
+async function postGovernmentEntityDocument (req, res, next) {
+  const isSwitchingCredentials = isSwitchingCredentialsRoute(req)
+  const collectingAdditionalKycData = isAdditionalKycDataRoute(req)
+  const currentCredential = getCurrentCredential(req.account)
+
+  const stripeAccountSetup = req.account.connectorGatewayAccountStripeProgress
+
+  if (!stripeAccountSetup) {
+    return next(new Error('Stripe setup progress is not available on request'))
+  }
+  if (stripeAccountSetup.governmentEntityDocument) {
+    const errorPageData = getAlreadySubmittedErrorPageData(req.account.external_id,
+      'You’ve already provided government entity document. Contact GOV.UK Pay support if you need to update it.')
+    return response(req, res, 'error-with-link', errorPageData)
+  }
+
+  let errors
+  const file = req.file
+
+  errors = validateFile(file)
+
+  if (!lodash.isEmpty(errors)) {
+    return response(req, res, 'stripe-setup/government-entity-document/index', {
+      isSwitchingCredentials,
+      errors
+    })
+  } else {
+    try {
+      const stripeAccountId = await getStripeAccountId(req.account, isSwitchingCredentials, req.correlationId)
+
+      const stripeFile = await uploadFile(`entity_document_for_account_${req.account.gateway_account_id}`, file.mimetype, file.buffer)
+      await updateAccount(stripeAccountId, { entity_verification_document_id: stripeFile.id })
+
+      await connector.setStripeAccountSetupFlag(req.account.gateway_account_id, 'government_entity_document', req.correlationId)
+
+      logger.info('Government entity document uploaded for Stripe account', {
+        stripe_account_id: stripeAccountId,
+        is_switching: isSwitchingCredentials,
+        collecting_additional_kyc_data: collectingAdditionalKycData
+      })
+
+      if (isSwitchingCredentials) {
+        return res.redirect(303, formatAccountPathsFor(paths.account.switchPSP.index, req.account.external_id))
+      } else if (collectingAdditionalKycData) {
+        const taskListComplete = await isKycTaskListComplete(currentCredential)
+        if (taskListComplete) {
+          await completeKyc(req.account.gateway_account_id, req.service, stripeAccountId, req.correlationId)
+          req.flash('generic', 'You’ve successfully added all the Know your customer details for this service.')
+        } else {
+          req.flash('generic', 'Document has been submitted successfully')
+        }
+        return res.redirect(303, formatAccountPathsFor(paths.account.yourPsp.index, req.account && req.account.external_id, currentCredential.external_id))
+      }
+
+      return res.redirect(303, formatAccountPathsFor(paths.account.stripe.addPspAccountDetails, req.account && req.account.external_id))
+    } catch (err) {
+      next(err)
```

Comment:
I've tested this locally, and multer breaks the AsyncLocalStorage session we use for logging with useful info such as the gateway_account_id, user_external_id etc.

Getting
```
{
  "logger_name": "/Users/stephendaly/projects/gds/pay-selfservice/app/controllers/stripe-setup/government-entity-document/post.controller.js",
  "message": "Logging from the POST controller",
  "level": "INFO",
  "@timestamp": "2021-12-01T16:18:41.639Z",
  "@version": 1,
  "container": "selfservice"
}
```

When should be getting something like 
```
{
  "logger_name": "/Users/stephendaly/projects/gds/pay-selfservice/app/controllers/stripe-setup/government-entity-document/get.controller.js",
  "message": "Logging from the GET controller",
  "level": "INFO",
  "user_external_id": "136f59fa302843dfa38618169b73cdf4",
  "internal_user": false,
  "gateway_account_id": 10,
  "gateway_account_type": "live",
  "service_external_id": "c7dcda05ae444e118ec1e5b216ce01ae",
  "@timestamp": "2021-12-01T16:18:14.632Z",
  "@version": 1,
  "container": "selfservice"
}
```

This means that in particular, we won't get the `x_request_id` on log lines. I think we can make this better this for this route by surrounding the whole controller with a try/catch block and manually logging an error line with all the additional details we expect before passing the error to next. Then at least if there is an error we can link it back to a request and an account.